compiler/src/jdk.graal.compiler.test/src/jdk/graal/compiler/core/test/ExceptionHandlerReachabilityTest.java

```diff
@@ -100,36 +100,40 @@ public byte[] generateClass(String className) {
-                                                                            catchBlock -> catchBlock.catching(cd(IllegalArgumentException.class), b -> b
-                                                                                            .labelBinding(handlerEx1)
-                                                                                            .iload(1)
-                                                                                            .invokestatic(classSharedExceptionHandlerClass, "baz", MethodTypeDesc.of(CD_int, CD_int))
-                                                                                            .istore(1)
-                                                                                            .iload(1)
-                                                                                            .ireturn()))
-                                                            .trying(tryBlock -> tryBlock
-                                                                            .iload(1)
-                                                                            .invokestatic(classSharedExceptionHandlerClass, "bar", MethodTypeDesc.of(CD_int, CD_int))
-                                                                            .istore(1),
-                                                                            catchBlock -> catchBlock.catching(cd(NumberFormatException.class), b -> b
-                                                                                            .iload(1)
-                                                                                            .invokestatic(classSharedExceptionHandlerClass, "doSomething",
-                                                                                                            MethodTypeDesc.of(CD_int, CD_int))
-                                                                                            .istore(1)
-                                                                                            .goto_(handlerEx1)))
-                                                            .iload(1)
-                                                            .ireturn();
-                                        })));
+                        .withMethodBody("sharedExceptionHandlerMethod", MethodTypeDesc.of(CD_int, CD_int), ACC_PUBLIC_STATIC, b -> {
+                            Label handlerEx1 = b.newLabel();
+                            b
+                                            .iload(0)
+                                            .istore(1)
+                                            .trying(tryBlock -> {
+                                                tryBlock
+                                                                .iload(1)
+                                                                .invokestatic(classSharedExceptionHandlerClass, "foo", MethodTypeDesc.of(CD_int, CD_int))
```

Comment:
You can cache this MethodTypeDesc in a variable and reuse that variable. MethodTypeDesc caches its descriptor string, so reusing the same instance is actually helpful to the ClassFile API's generation.